src/solvers/smt2/smt2_conv.cpp

```diff
@@ -5127,8 +5127,9 @@ void smt2_convt::convert_type(const typet &type)
     // we always use array theory for top-level arrays
     const typet &subtype = array_type.element_type();
 
+    // Arrays map the index type to the element type.
     out << "(Array ";
-    convert_type(array_type.size().type());
+    convert_type(array_type.index_type());
```

Comment:
Wasn't `.size().type()` correct here?

Reply:
It only works when the array has a size.

Reply:
What information does SMT-LIB require at this point? Agreed that not all arrays have a size, which may be a problem, but I'm just not sufficiently knowledgable in SMT-LIB to tell whether this change really makes sense. If that's just a workaround for a missing size then I remain unconvinced this is the right change to make.

Reply:
SMT-LIB does require the index type; arrays are maps from indices to the codomain type. Using the type of the size was a workaround.